service/src/main/java/org/jboss/sbomer/service/scheduler/GenerationRequestScheduler.java

```diff
@@ -0,0 +1,174 @@
+import org.jboss.sbomer.service.leader.LeaderManager;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.scheduler.Scheduled;
+import io.quarkus.scheduler.Scheduled.ConcurrentExecution;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.transaction.Transactional.TxType;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class GenerationRequestScheduler {
+
+    SbomGenerationRequestRepository requestRepository;
+
+    KubernetesClient kubernetesClient;
+
+    GenerationSchedulerConfig generationSchedulerConfig;
+
+    LeaderManager leaderManager;
+
+    @Inject
+    public GenerationRequestScheduler(
+            SbomGenerationRequestRepository requestRepository,
+            KubernetesClient kubernetesClient,
+            GenerationSchedulerConfig generationSchedulerConfig,
+            LeaderManager leaderManager) {
+        this.requestRepository = requestRepository;
+        this.kubernetesClient = kubernetesClient;
+        this.generationSchedulerConfig = generationSchedulerConfig;
+        this.leaderManager = leaderManager;
+    }
+
+    /**
+     * <p>
+     * Schedules a batch of generations to be handled in the current deployment (if there is capacity available).
+     * </p>
+     *
+     * <p>
+     * If there are less than {@link GenerationSchedulerConfig#maxConcurrentGenerations()} generations currently running
+     * it will check whether there are some generations waiting to be scheduled in the database. If there are such
+     * generations found, it will fetch up to {@link GenerationSchedulerConfig#syncBatch()} generations and create
+     * {@link GenerationRequest} instances (basically ConfigMap) for each one of them. These ill then will be picked by
+     * the controller.
+     * </p>
+     *
+     *
+     * <p>
+     * This method is run periodically. By default every 15 seconds. It is controlled by the
+     * {@code sbomer.service.generation-scheduler.sync-interval} property.
+     * </p>
+     */
+    @Scheduled(
+            every = "${sbomer.service.generation-scheduler.sync-interval:15s}",
+            delay = 1,
+            delayUnit = TimeUnit.MINUTES,
+            concurrentExecution = ConcurrentExecution.SKIP)
+    @Transactional(value = TxType.REQUIRES_NEW)
+    public void scheduleGenerations() {
+        if (!leaderManager.isLeader()) {
+            log.info("Current instance is not the leader, skipping scheduling of generations in this instance");
+            return;
+        }
+
+        // Get all ConfigMaps that represent generation requests within the namespace that are in progress
+        int scheduledGenerationsCount = kubernetesClient.configMaps()
+                .withLabelSelector(
+                        "sbomer.jboss.org/type=generation-request,sbomer.jboss.org/status notin (FAILED, FINISHED)")
```

Comment:
We could probably add the NO_OP which is just a placeholder and effectively does not require a generation with computation. But those would be very few numbers anyway.

Reply:
Yah, I think these can be ignored. What we need to do is to keep better track of what is happening in a particular cluster. Ghost CM's could lead to not scheduling anything.

Hm, now that I think of it -- we could have even a feature flag to control whether things should be scheduled at all! Even per cluster. But it's a future improvment.